### PR TITLE
Ad review

### PR DESCRIPTION
**Pull request by ciseng**
Each commit in this pull request handles one outstanding issue. Specifically: (1) Made DTLS profile normative, clarified exceptions to adopting that draft (2) Made JSON array base64-encoded to be used as a single scope value; (3) explained retained messges).

### CODE REVIEW

File: draft-ietf-ace-mqtt-tls-profile.xml [hunk 1]
```diff
@@ -461,38 +463,43 @@
                 from the AS. The Client and the AS MUST perform mutual authentication.
                 The Client requests an access token from the AS as
                 described in Section 5.8.1 of the <xref target="I-D.ietf-ace-oauth-authz">ACE framework</xref>. 
-                In order to use an asymmetric key (i.e., Raw Public Key), the client MUST add a "req_cnf" object 
-                carrying either its raw public key or a unique identifier for a public key that it has 
-                previously made known to the AS. The media type is 'application/ace+json'.
-                The AS uses JSON in the payload of its responses to the Client and the RS.
+                The document follows the procedures defined for in Section 3.2.1 of the <xref target="I-D.ietf-ace-dtls-authorize">DTLS profile</xref>
```
Comment (kaduk) [inline draft-ietf-ace-mqtt-tls-profile.xml:466]:
```suggestion
                The document follows the procedures defined in Section 3.2.1 of the <xref target="I-D.ietf-ace-dtls-authorize">DTLS profile</xref>

```

File: draft-ietf-ace-mqtt-tls-profile.xml [hunk 2]
```diff
@@ -461,38 +463,43 @@
...
-                in Section 5.8.2 of <xref target="I-D.ietf-ace-oauth-authz">the ACE framework</xref>, and
-                specifically, the "ace_profile" parameter is set to "mqtt_tls".
+                the AS issues an access token (Figure 1 (B)). 
+            </t>
+            <t>The response includes the parameters described
+                in Section 5.8.2 of <xref target="I-D.ietf-ace-oauth-authz">the ACE framework</xref>.
+                For RPK, the parameters are as described 
+                in Section 3.2.1 of the <xref target="I-D.ietf-ace-dtls-authorize">DTLS profile</xref>. 
+                For PSK, the document follows Section 3.3.1 of the <xref target="I-D.ietf-ace-dtls-authorize">DTLS profile</xref>.
+                In both cases,  if the response MUST contain an "ace_profile" parameter, this parameter is set
```
Comment (kaduk) [inline draft-ietf-ace-mqtt-tls-profile.xml:483]:
```suggestion
                In both cases,  if the response contains an "ace_profile" parameter, this parameter is set 

```

File: draft-ietf-ace-mqtt-tls-profile.xml [hunk 3]
```diff
@@ -597,60 +608,54 @@
...
+                  <xref target="I-D.ietf-ace-extend-dtls-authorize"></xref>. 
+                When there are exceptions to the DTLS profile, these are explicitly 
+                stated in the document.
+                If TLS 1.2 is used, <xref target="RFC7925"></xref> describes 
                 how TLS can be used for constrained devices, alongside recommended cipher suites. 
-                Additionally, 
-                TLS 1.2 implementations SHOULD use "Extended Master Secret" <xref target="RFC7627"></xref>
-                to include handshake transcript into the master secret (RFC 7627 describes alternative conditions that must be met to defend against the same attacks without the extension).  TLS implementations SHOULD use the
+                Additionally, TLS 1.2 implementations MUST use "Extended Master Secret" <xref target="RFC7627"></xref>
+                to include handshake transcript into the master secret.  TLS implementations SHOULD use the
```
Comment (kaduk) [inline draft-ietf-ace-mqtt-tls-profile.xml:622]:
```suggestion
                to incorporate the handshake transcript into the master secret.  TLS implementations SHOULD use the

```

File: draft-ietf-ace-mqtt-tls-profile.xml [hunk 4]
```diff
@@ -597,60 +608,54 @@
...
+                 in the "cnf" structure and used as a PSK identity in TLS.
+                 As an alternative to the access token upload, the client can provide the most recent access token, 
+                 JWT or CWT, as a PSK identity. 
                 </t>
                 <t>
-                In PreSharedKey mode, the knowledge of the shared secret by the
-                Client and the Broker is used for mutual authentication
-                between both peers.
+                In contrast to <xref target="I-D.ietf-ace-dtls-authorize">DTLS profile for ACE</xref>, 
+                a client MAY NOT support the cipher suite TLS_PSK_WITH_AES_128_CCM_8, but
```
Comment (kaduk) [inline draft-ietf-ace-mqtt-tls-profile.xml:657]:
```suggestion
                a client MAY omit support for the cipher suite TLS_PSK_WITH_AES_128_CCM_8, but

```

File: draft-ietf-ace-mqtt-tls-profile.xml [hunk 5]
```diff
@@ -959,25 +964,27 @@
...
+                 During the CONNECT, if the Will Flag is set to 1, the Broker MUST also authorize
+                the publication of the Will Topic and message using the token's scope field.
+            The Broker uses the scope to match against the Topic Name in a PUBLISH packet
             (including Will Topic in the CONNECT) or a Topic Filter in a SUBSCRIBE packet.
             </t>
             <t>
             The scope in the token is a single value. For a JWT, the single scope 
-            has an internal structure of a JSON array, and for a CWT, this information is 
-            represented in CBOR following the <xref target="I-D.ietf-ace-aif">Authorization Information 
+            is base64 encoded string, which has an internal structure of a JSON array.
```
Comment (kaduk) [inline draft-ietf-ace-mqtt-tls-profile.xml:978]:
```suggestion
            is base64url encoded string with any padding characters removed, which has an internal structure of a JSON array. 

```

File: draft-ietf-ace-mqtt-tls-profile.xml [hunk 6]
```diff
@@ -1753,8 +1774,9 @@ those specified for the 'application/json' media type.</t>
 
 	<section anchor="Acknowledgments" title="Acknowledgments" numbered="no" toc="default">
 		<t>
-                The authors would like to thank Ludwig Seitz for his review and his input on the authorization information endpoint.
-                The authors would like to thank Paul Fremantle for the initial discussions on MQTT v5.0 support. 
+                The authors would like to thank Ludwig Seitz for his review and his input on the authorization information endpoint, 
+                and Benjamin Kadul for his review, insightful comments, and contributions to resolving issues.
```
Comment (kaduk) [inline draft-ietf-ace-mqtt-tls-profile.xml:1778]:
```suggestion
                and Benjamin Kaduk for his review, insightful comments, and contributions to resolving issues.

```

**Review by kaduk — APPROVED** [review]
This seems good enough to merge and upload

File: draft-ietf-ace-mqtt-tls-profile.xml (hunk 4 above)
Comment (ciseng) [reply draft-ietf-ace-mqtt-tls-profile.xml:660]: I see that I haven't noticed this suggestion but accepted this suggestion as a new commit in the main branch.

**Pull request merged**